lua/neo-tree/ui/renderer.lua

```diff
@@ -315,7 +316,13 @@ end
 
 local prepare_node = function(item, state)
   if item.skip_node then
-    return nil
+    if item.is_empty_with_hidden_root then
+      local line = NuiLine()
+      line:append("(empty folder)", highlights.MESSAGE)
```

Comment:
> Your solution is completely bypassing the Nui Nodes that the rest of Neo-tree and any third party plugins use

Would setting the type of the node to `message` here help resolve the issue? I couldn't figure out how to do that, the text just ended up with normal highlighting when I do `item.type = "message"` and return `"(empty folder)"`

Reply:
Ya setting the type to message [should work](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/v2.x/lua/neo-tree/defaults.lua#L274-L277). When you say 

> return `"(empty folder)"`

What do you mean?
I'm unsure we want to be overwriting the type that is passed in. 

A side note, if we completely remove the changes that are being proposed to the renderer, your solution still works. I don't know that we need to be touching renderer at all (aside from the nil bug I mentioned earlier, though I can submit a PR for that if we want as its outside the scope of this issue).

Reply:
> What do you mean?

with this patch, `(empty folder)` doesn't highlight as a message, probably because `item.type = "message"` doesn't do what I think it does. If there is some way to change the `type` to `"message` inside `prepare_node`, would it address your concerns about `bypassing the Nui Nodes`?

```patch
--- a/lua/neo-tree/ui/renderer.lua
+++ b/lua/neo-tree/ui/renderer.lua
@@ -317,9 +317,8 @@ end
 local prepare_node = function(item, state)
   if item.skip_node then
     if item.is_empty_with_hidden_root then
-      local line = NuiLine()
-      line:append("(empty folder)", highlights.MESSAGE)
-      return line
+      item.type = "message"
+      return "(empty folder)"
     else
       return nil
     end
```

> if we completely remove the changes that are being proposed to the renderer, your solution still works

On my end, the hang is gone, but the created file ends up in `foo_empty_message` instead of `foo`. I am not experiencing the nil bug, if you can't reproduce this I can create a minimal reproducible example.

Reply:
It seems I got lost while looking at the PR, I was getting the changes in `prepare_node` and `show_nodes` mixed in my head lol. In this case, item.type isn't going to be used as I referenced above. Disregard my current complaint about `prepare_node` changes :)